### PR TITLE
Add food_* groups to default edibles

### PR DESCRIPTION
**Pull request by tenplus1**
This adds the food_* groups to edible items within default so that food crafting is easier within mods.

### CODE REVIEW

File: mods/farming/init.lua [hunk 1]
```diff
@@ -25,17 +25,21 @@ farming.register_plant("farming:wheat", {
 	place_param2 = 3,
 })
 
+minetest.override_item("farming:wheat", {
+	groups = {food_wheat = 1, flammable = 2},
```
Comment (paramat) [inline mods/farming/init.lua]: Wheat wouldn't be used in a food recipe, only flour.

**Comment by tenplus1** [comment]
You'd be surprised how many recipes ask for wheat as an item.

**Comment by paramat** [comment]
Ok.

**Comment by Ezhh** [comment]
Did comment earlier in -hub that I don't like the override. That's really the only thing putting me off this. A mod needing to override what it registers itself is bad.

**Comment by paramat** [comment]
I'm also unsure about the whole PR, undecided.

**Comment by Ezhh** [comment]
I'm fine with this PR, if the override issue is resolved. 

We've previously talked about adding more groups to help enable mods to be less game specific, and this seems like a good step toward that.

**Comment by tenplus1** [comment]
The only other way to do this is to rewrite the farming api for registering crops so that it adds custom groups, and that overcomplicates this whole issue.

**Comment by tenplus1** [comment]
Okies, when using farming.register_plant() it will not use 'def.groups' for the output crop (wheat, cotton etc.) so the override is no longer required :)

**Comment by paramat** [comment]
Much better.
I'm neutral on the PR.

**Comment by tenplus1** [comment]
@paramat you could +1 it simply for the fact it fixes the farming custom groups in the api.

**Comment by paramat** [comment]
@tenplus1 i could also be neutral.

**Comment by Fixer-007** [comment]
You are neutral because unsure, or there are negative side effects?

**Comment by paramat** [comment]
I can't see any negatives (apart from code bloat / complexity), just unsure if useful enough to be justified. Other devs can decide.

**Pull request merged**